### PR TITLE
Make SPO validate configuration before doing custom validation

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Found a small missing piece of code: when SPO connector runs validation, it never does backend validation of the configuration (validation of field types, required fields and so on).

This PR adds this logic to make sure basic validation happens as well.

### CODE REVIEW

File: connectors/sources/sharepoint_online.py [hunk 1]
```diff
@@ -946,6 +946,8 @@ def get_default_configuration(cls):
         }
 
     async def validate_config(self):
+        self.configuration.check_valid()
```
Comment (wangch079) [inline connectors/sources/sharepoint_online.py:949]:
```suggestion
        super().validate_config()
```

I think this pattern can also apply to other sources that are using `self.configuration.check_valid()`

Comment (artem-shelkovnikov) [reply connectors/sources/sharepoint_online.py:949]: Yeah but for now I chose the consistent call here - same as in other connectors.

TBH I also think that `configuration.check_valid()` should be called outside - I think all connectors need to have valid configuration to function. If we follow current pattern, some connectors can by mistake or on purpose ignore it and allow invalid config, which is just weird. I think we should revisit this.

Comment (artem-shelkovnikov) [reply connectors/sources/sharepoint_online.py:949]: What do you think if we merge it like it is now and make a tech debt issue to address this?

**Review by navarone-feekery — APPROVED** [review]

File: connectors/sources/sharepoint_online.py (hunk 1 above)
Comment (navarone-feekery) [reply connectors/sources/sharepoint_online.py:949]: ++ on making a tech debt issue

Comment (artem-shelkovnikov) [reply connectors/sources/sharepoint_online.py:949]: Created an issue: https://github.com/elastic/connectors-python/issues/1194

**Pull request merged**